Helper/Entity/ProductHelper.php

```diff
@@ -20,19 +21,22 @@
 use Magento\Catalog\Model\Product\Type\AbstractType;
 use Magento\Catalog\Model\Product\Visibility;
 use Magento\Catalog\Model\ResourceModel\Eav\Attribute as AttributeResource;
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
```

Comment:
I think moving this into the constructor could introduce a subtle change. Previously `objectManager->create()` would provide a new Collection object every time it's run. Moving it into the constructor would make it a shared object. I don't have any Algolia instance to test but from looking at the code using `CollectionFactory` and then `->create()` on it seems safer.

Reply:
Thanks, you're right! I changed it to use the factory.